### ISSUE TITLE
Migração CPF atributo para USERNAME

### ISSUE THREAD

**Issue by vicmagpac** [EscolaDeSaudePublica/idsaude-api#4]
## **Objetivo**

**Como** usuária do ID Saúde  
**Quero** acessar o ID Saúde depois da autenticação realizada pelo CPF ou EMAIL  
**Para** acessar os recursos internos da minha conta

## **Contexto**

- Atualmente o ID Saúde a forma de autenticação está sendo feita apenas pelo email. O CPF está nos campos de atributos do usuário, o que permite o mesmo internamente ser duplicado. Para resolver o problema de duplicação e garantir consistência do usuário internamente, é necessário usar o CPF ou EMAIL ao realizar cadastro e login.

## **Escopo**

- [x] Migrar CPF atributos do usuário para USERNAME

## **Fora de Escopo**

- migrar qualquer outro atributo

## **Critérios de Aceitação**

- [x] Migrar CPF atributos do usuário para USERNAME  
**Dado que** acesso todos os usuários  cadastrados no IDSaúde
**Quando** acesso cada atributo CPF do usuário  
**então** realizo a atualização do registro cpf -> username (único)

# Observações

execução do comando
acessar o container
`docker exec -it HASH_CONTAINER bash`

Comando artisan
`php artisan idsaude:migra-cpf-username`

A seguir, verificar no idsaúde a migração.

**Comment by ericsonmoreira** [comment]
@victorMagalhaesPacheco tu poderia dar o contexto dessa atividade para a equipe do iSUS.

**Comment by vicmagpac** [comment]
> @victorMagalhaesPacheco tu poderia dar o contexto dessa atividade para a equipe do iSUS.

Sim! Podemos realizar marcar. A explicação já foi feita pra equipe do mapas, importante realizarmos o alinhamento também com a equipe do iSUS.

**Comment by uira** [comment]
@victorMagalhaesPacheco Aproveita e chama o Wandson e @jacsonp

**Comment by Laysacunha** [comment]
@victorMagalhaesPacheco essa á a integração que estamos alinhando nas ultimas semanas?

**Comment by ricardopoppi** [comment]
@marameida aqui também precisamos do acesso ao ambiente dev do iSUS (homologação) para homologar

**Comment by marameida** [comment]
Ciente. @ricardopoppi podemos ver com equipe do ISUS na reunião agendada para amanhã (10h).

**Comment by lucaspq** [comment]
Houve uma reunião de alinhamento com @victorMagalhaesPacheco @ericsonmoreira para subir essa funcionalidade para produção, no entanto encontramos algumas pendências do lado do iSUS e Mapa para essa integração. Assim que @ericsonmoreira implementar e testar essa mudanças, subimos para produção.
